test/unit/lexer/token/equality.d

```diff
@@ -0,0 +1,349 @@
+module lexer.token.equality;
```

Comment:
A new file, 350 lines, no documentation, doing I have little idea what.

Reply:
Testing all the tokens. I can add some documentation.

Reply:
It will be slimed down when the lexer switches the diagnostic handler from an abstract class to a delegate, as we discussed at DConf.

Reply:
Added documentation.